saleor/core/exceptions.py

```diff
@@ -73,3 +75,7 @@ def __init__(self, message: str):
         super().__init__(message)
         self.message = message
         self.code = CheckoutErrorCode.GIFT_CARD_NOT_APPLICABLE.value
+
+
+class CircularQuery(GraphQLError):
```

Comment:
Is `Query` correct? Without checking where it is used I am not sure that I would guess what is this.
The issue is related only to `Subscription` + `Sync Event`, right? So maybe something like `CircularSubscriptionQuery` or `CircularSubscriptionSyncEvent`? Or something else that will give more  context than only `Query`.

Reply:
'CircularSubscriptionSyncEvent' sounds good :)